### PR TITLE
[DON'T MERGE] Marketplace integration

### PR DESCRIPTION
**Pull request by QuentinI**

### CODE REVIEW

File: marketplace-solver/src/api.rs [hunk 1]
```diff
@@ -101,8 +102,16 @@ where
     api.post("submit_bid", |_req, _state| {
         async move { Ok("Bid Submitted") }.boxed()
     })?
-    .get("auction_results", |_req, _state| {
-        async move { Ok("Auction Results Gotten") }.boxed()
+    .get("auction_results", |req, _state| {
+        async move {
```
Comment (imabdulbasit) [inline marketplace-solver/src/api.rs:106]: I forgot to call the `calculate_auction_results_permissionless` from the state here. I can push the change if you are okay w that

File: marketplace-solver/src/api.rs [hunk 2]
```diff
@@ -101,8 +102,16 @@ where
...
+            Ok(SolverAuctionResults::new(
+                ViewNumber::new(view_num),
+                vec![],
+                vec![(10_000_u32.into(), "http://localhost:31003".parse().unwrap())],
+            ))
+        }
+        .boxed()
     })?
     .get("auction_results_permissioned", |_req, _state| {
         async move { Ok("Permissioned Auction Results Gotten") }.boxed()
```
Comment (imabdulbasit) [inline marketplace-solver/src/api.rs:117]: same here `calculate_auction_results_permissioned`

File: marketplace-solver/src/api.rs (hunk 1 above)
Comment (imabdulbasit) [reply marketplace-solver/src/api.rs:106]: or I think I'll create a separate PR for it and merge it into main

Comment (imabdulbasit) [reply marketplace-solver/src/api.rs:106]: https://github.com/EspressoSystems/espresso-sequencer/pull/1843

File: types/src/v0/impls/auction.rs [hunk 3]
```diff
@@ -291,10 +292,14 @@ impl<TYPES: NodeType> AuctionResultsProvider<TYPES> for SolverAuctionResultsProv
         &self,
         view_number: TYPES::Time,
     ) -> anyhow::Result<TYPES::AuctionResult> {
-        let resp = SurfClient::new(self.0.clone())
-            .get::<TYPES::AuctionResult>(&format!("/v0/api/auction_results/{}", *view_number))
-            .send()
-            .await?;
+        let resp = SurfClient::new(
```
Comment (tbro) [inline types/src/v0/impls/auction.rs:295]: This works as so? I'll make a PR for sequencer then. Though I'm wondering if we shouldn't have these paths in configuration.

Comment (tbro) [reply types/src/v0/impls/auction.rs:295]: PR https://github.com/EspressoSystems/espresso-sequencer/pull/1853

Comment (elliedavidson) [reply types/src/v0/impls/auction.rs:295]: Yes, +1 to putting the paths into a config somewhere!

Comment (tbro) [reply types/src/v0/impls/auction.rs:295]: I'll do a separate PR

**Pull request merged**